src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -2553,6 +2553,7 @@ private void VisitBinaryOperatorChildren(BoundBinaryOperator node)
         protected virtual void VisitBinaryOperatorChildren(ArrayBuilder<BoundBinaryOperator> stack)
         {
             var binary = stack.Pop();
+            EnterRegionIfNeeded(binary);
```

Comment:
We don't call 'VisitAlways' on the binary nodes themselves to avoid recursion. Therefore we need to take care to call Enter/LeaveRegion APIs appropriately when we start/finish with those nodes.

See also #74317 for a non-recursive visitor which makes similar calls.